### DISCUSSION TITLE
pinout / components for the cav-mn10?

### DISCUSSION THREAD

**Discussion by arrjay** [4gra/pclk-mn10#6] [General]
actually playing (shortly) with one of the pc3 units, and trying to look at what driving it without the cav-mn10 might be, but it looks like no one has covered a possible pinout - do you have any references to whats inside the thing?

**Comment by 4gra** [comment]
First of all, let me tell you how excited I am that someone else on the planet is still using one of these things 😁 TL;DR: I didn't have to deal with any electrical or pc-link bus detail thanks to the MN10 unit, but I did lots of reading ... and turned up very little. 

I know that what's inside is logically completely split: the USB device is two devices in one: a USB sound card, and a USB "bulk transfer" (standard serial I/O) device.  The two functions are entirely independent. So as far as the PC3 goes, one only needs to connect the "PC LINK" portion to get control of the device.

The manual (https://www.sony.co.uk/electronics/support/res/manuals/4230/42305744M.pdf) calls it a "PC LINK (Mini DIN 6P) connector".  It certainly is a 6-pin mini-DIN, as you can see from the back of the unit. Cracking it open all six pins are wired at the other end (plus a seventh, with extra insulation, assumed ground). The circuit board doesn't tell _me_ anything more - the wires are numbered 1-7 but otherwise unlabelled so far as I can see, and I'm not qualified to trace anything useful from it, nor have I the equipment to inspect what's going on electrically without damage.  However I can post photos of the circuit board (heavily complicated by the optical and analogue outputs, no doubt) if they'd be of use, and if you can propose something useful I could do with a multimeter then I could try it out.

Trying to guess what it might be, the MN10 commands bear more than a passing resemblance to S-Link / Control-A1 system (with more padding).  But this is of no use to you electrically of course and, browsing the internet, the "PC LINK" is unusual in most respects, so there's no reason to assume it would be the same as other link protocols.  I was briefly excited to see official reference to a 6-pin DIN connector in a Sony manual for the PCLK-PX1, a similar link device described as an "RS-232 to Control-I" adaptor (https://www.sony.ee/electronics/support/res/manuals/3865/38650061M.pdf).  Any suggestion of similarity is refuted by another official page for the PCLK-MN20 (still just a CAV-MN10) https://www.sony.jp/products/musicnetwork/contents/pclk-mn20/ which explicitly states "The PCLINK terminal of [the PCLK-MN10] can not be connected to CONTROL I terminal, control A1 terminal, control A1II terminal. Please use PCLK-PX3 for equipment with CONTROL I terminal.".  And in any event there's virtually no information on how Control-I was meant to work.

So there's a possibility that the PC3 really was a one-off... or perhaps it's just the usual jumbled set of Sony codes in a slightly different package.  Please do let me know if you find anything out, I'd be delighted to add more information to the wiki here.

**Comment by arrjay** [comment]
I do think they're very closely related - http://forums.sonyinsider.com/topic/29182-ctrla1-to-pc-link-mod/ is talking about flipping a player from one to the other (unfortunately, it doesn't note if things worked!) and http://forums.sonyinsider.com/topic/29130-adding-pc-link-to-a-jb930/ seems to _imply_ pc-link is also related to ps/2 or some units used the mini-din for both functions. I'll take apart the pc3 when I get it and at least grab pictures of the insides.

**Comment by arrjay** [comment]
and yeah, if it's really just serial wrapped around bit shifting, http://www.brian-patti.com/s-link/CAVuser.htm really feels like a good step to build on top of, perhaps with a modified cable to sit between the two and grab the electrical signaling. it looked like the CAV side is a mini-din plug?

**Comment by 4gra** [comment]
It's definitely a mini-DIN, 6-pin. The link you posted about the conversion, which says they both use SIRCS, is very hopeful. I'll post photos when I can get good lighting.

(skipping past the tricky signalling bits...) I have a theory that the MN10 is just truncating the input slightly and sending out the rest of the message verbatim.

**Comment by arrjay** [comment]
fell down another hole looking for protocol information, and this just seems to conveniently similar to ignore - it's sony's protocol for controlling cameras!
http://www.boehmel.de/lanc.htm <- potential pinout?
https://en.wikipedia.org/wiki/LANC
https://www.hackster.io/L-Rosen/serial-to-lanc-control-l-70f735

**Comment by 4gra** [comment]
That looks like a 5-pin DIN so I'm not sure how related the signalling is likely to be - though of course LANC looks like yet another variant of much the same sort of message passing system. Did you see the references at the bottom of the main wiki page: https://github.com/4gra/pclk-mn10/wiki ?

**Comment by arrjay** [comment]
I got the pc3 and confirm it works, I'll have some pictures later, I also dug up the service schematics and paged through them finally (as they are right on http://www.minidisc.org/manuals/sony/service/sony_MDS-PC3_service_manual.pdf ) and on page 37 the pc-link din is clearly identified...as Control-I! We have a pinout!

**Comment by 4gra** [comment]
That's a great find.  So it looks like it's pure "Control-I" but with an extra line to control power on/off?

BTW I can confirm this pinout (and protocol) is common to the MDS-PC3, DHC-MD595 and MDS-JE470; all will work with the PCLK-MN10.  So it's definitely worth figuring out how Control-I relates to the signalling protocol, because this would presumably allow us to (a) do without a PCLK-MN10 as you're suggesting but also (b) adapt the PCLK-MN10 to work with *all* Control-I devices (barring the power on command).

<img width="170" alt="control-i" src="https://user-images.githubusercontent.com/157833/49700926-2ca2a180-fbdd-11e8-95e6-83a13f1ebd31.png">

**Comment by 4gra** [comment]
Just looking back at this... thanks again for getting in touch. Did you make concrete progress?

Hardware is a little out of scope, but if my tool could be used with another tranche of machines with little more than an adaptor, or an arduino or something, I'll happily add links or documentation.

**Comment by arrjay** [comment]
well

I finally got the counterpart CAV-MN10

so I intend to find out!

**Comment by Sir68k** [comment]
I successfully managed to control my J300 with PC-Link using I2C on an ESP32. I made a project to document all this: https://github.com/Sir68k/pclink-i2c/ . Hope you find it helpful!

**Reply by 4gra** [reply]
That's really cool, first of all because this means I won't run out of MN-10s but also because I think it could help us better understand the "address" bytes that prefix the actual commands.  I note for example the difference in the 'power on' command https://github.com/4gra/pclk-mn10/wiki/messages suggesting that this command never reaches the main board, and instead uses the PCLK power mechanism identified in your repo.

**Reply by Sir68k** [reply]
If you want to do a quick test, you could see if "byte 2: always zeros" can be replaced with 0x09 for MD commands, and 0x10 for all others. If so, that means that this is just the I2C address.

**Reply by Sir68k** [reply]
Also, because I2C commands are used internally as well, there are many undocumented commands showing up (e.g. for writing text to the display, other commands for ejecting the CD as well, ...). It's quite fun, experimenting with it.

**Reply by 4gra** [reply]
I'm so sorry I somehow missed these messages, thanks so much for the comments @Sir68k!

Byte 2 turns out to be used exclusively for "continuation" messages on the PCLK protocol.  The address bytes follow afterwards. 

And if you find any other useful messages experimentally, please feel free to share them -- I'm *particularly* interested in writing text to the display!

**Comment by arrjay** [comment]
I did pick this up again - and am at least recording stuff as I go along here: https://github.com/arrjay/pc3search/blob/main/nodisc-i2c.txt

I'm also doing some very initial playing with the netmd-js code (it is very _not_ netmd-js) here: https://github.com/cybercase/netmd-js/compare/master...arrjay:netmd-js:master and puzzling through some of it's state machine. mine (with the pc3) at least seems to want continual polling on the read input to really *work*

**Reply by 4gra** [reply]
I definitely find this on my primary combination of mn10/md595. If the read buffer isn't reasonably clear, things can wedge. Interesting that it should care because there's rarely any need to acknowledge or reference a previous command. So turning on the clock, or control while a CD/MD is playing, tends to encourage more reads.

My md595 is in quite a sorry state and MD experiments are probably out until I can work out the repair, but I've actually managed quite a bit more decoding recently by throwing more tooling at my original dumps. I need to push a newer display framework which shows changes better.  Watch this space...
